Add installIn parameter to SoloModule (#20)
JuulLabs/pommel: README.md

```kotlin
interface MyInterface

> @SoloModule(MyInterface::class, SingletonComponent::class)

Reviewer: Having these named would make it more clear what the parameters are for in the documentation.

```kotlin
interface MyInterface

> @SoloModule(MyInterface::class, SingletonComponent::class)

Reviewer: Same here.

Pommel expects `class`es annotated with `@SoloModule` to have a constructor that is is annotated with `@Inject`. Multiple constructors annotated with `@Inject` is not supported as Dagger itself only supports a single
constructor annotated with `@Inject`. All parameters passed into the constructor must also be on the Dagger graph, as this is another expectation of Dagger constructor injection

Pommel expects all functions annotated with `@SoloModule` to be `static` due to the fact that the generated module needs to call the annotated function.
As a result Pommel only supports top level functions and functions that are enclosed in an `object` or `companion` class

> Pommel was designed to be used with Dagger-Hilt as result it expects a component to be passed into the `installIn` parameter. The component passed into `installIn`  follows the same limitation of Dagger-Hilt.

Reviewer: ```suggestion
Pommel was designed to be used with Dagger-Hilt so it expects a component to be passed into the `installIn` parameter. The component passed into `installIn`  follows the same limitations imposed by Dagger-Hilt.
```
